From 09900b68289e2240dea8c7e2403620b931b18734 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Tue, 2 Sep 2025 16:36:18 -0700
Subject: [PATCH 5/5] Fix CMake targets to be relocatable

Generate a custom drjitConfig.cmake that handles relocatable paths
for conda-forge packaging. This ensures the CMake targets work
correctly when the package is installed in different locations.

Based on conda-forge patch 0005-Fix-cmake-targets-relocatable.patch
---
 CMakeLists.txt | 44 +++++++++++++++++++++++++++++++++++++++-----
 1 file changed, 39 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b5db4148..d06e6377 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -293,19 +293,53 @@ if (DRJIT_MASTER_PROJECT)
 
   set(DRJIT_CMAKECONFIG_INSTALL_DIR "${CMAKE_INSTALL_DATAROOTDIR}/cmake/drjit")
 
+  install(
+    EXPORT drjitTargets
+    DESTINATION ${DRJIT_CMAKECONFIG_INSTALL_DIR})
+
+  # Generate a custom drjitConfig.cmake that handles relocatable paths
+  set(DRJIT_CONFIG_CONTENT "
+@PACKAGE_INIT@
+
+get_filename_component(DRJIT_CMAKE_DIR \"\${CMAKE_CURRENT_LIST_FILE}\" PATH)
+if(WIN32)
+  get_filename_component(DRJIT_PREFIX \"\${DRJIT_CMAKE_DIR}/../..\" ABSOLUTE)
+else()
+  get_filename_component(DRJIT_PREFIX \"\${DRJIT_CMAKE_DIR}/../../..\" ABSOLUTE)
+endif()
+
+# Include the targets file but override paths to make them relocatable
+include(\"\${CMAKE_CURRENT_LIST_DIR}/drjitTargets.cmake\")
+
+# Fix absolute paths in imported targets for conda-forge packaging
+if(TARGET drjit-core)
+  if(WIN32)
+    set_target_properties(drjit-core PROPERTIES
+      IMPORTED_LOCATION \"\${DRJIT_PREFIX}/Library/bin/drjit-core.dll\"
+      IMPORTED_IMPLIB \"\${DRJIT_PREFIX}/Library/lib/drjit-core.lib\"
+    )
+  else()
+    set_target_properties(drjit-core PROPERTIES
+      IMPORTED_LOCATION \"\${DRJIT_PREFIX}/lib/libdrjit-core\${CMAKE_SHARED_LIBRARY_SUFFIX}\"
+    )
+  endif()
+endif()
+
+check_required_components(drjit)
+")
+
   configure_package_config_file(
-    resources/drjitConfig.cmake.in drjitConfig.cmake
+    "${CMAKE_CURRENT_BINARY_DIR}/drjitConfig.cmake.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/drjitConfig.cmake"
     INSTALL_DESTINATION ${DRJIT_CMAKECONFIG_INSTALL_DIR})
 
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/drjitConfig.cmake.in" "${DRJIT_CONFIG_CONTENT}")
+
   write_basic_package_version_file(
     drjitConfigVersion.cmake
     VERSION ${DRJIT_VERSION}
     COMPATIBILITY AnyNewerVersion ARCH_INDEPENDENT)
 
-  install(
-    EXPORT drjitTargets
-    DESTINATION ${DRJIT_CMAKECONFIG_INSTALL_DIR})
-
   install(
     FILES
     ${CMAKE_CURRENT_BINARY_DIR}/drjitConfigVersion.cmake
-- 
2.44.0.windows.1

